core/src/main/java/org/elasticsearch/index/analysis/AnalysisService.java

```diff
@@ -262,10 +261,13 @@ public AnalysisService(Index index, @IndexSettings Settings indexSettings, @Null
             }
         }
 
-        defaultAnalyzer = analyzers.get("default");
+        NamedAnalyzer defaultAnalyzer = analyzers.get("default");
         if (defaultAnalyzer == null) {
             throw new IllegalArgumentException("no default analyzer configured");
         }
+        if (analyzers.containsKey("default_index") && Version.indexCreated(indexSettings).onOrAfter(Version.V_2_0_0)) {
```

Comment:
If we are deprecating now, we either need to upgrade the mappings in master, or throw an error? The current code would allow still specifying the setting? I think we should error on new indexes (at minimum 3.0, although if we should error for 2.2+ or something, I dunno). But the deprecation warning should be for all indexes that still "support" this setting.